seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/reader/ReadStrategyFactory.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.connectors.seatunnel.file.source.reader;
+
+import org.apache.seatunnel.connectors.seatunnel.file.config.FileFormat;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class ReadStrategyFactory {
+
+    private ReadStrategyFactory() {}
+
+    public static ReadStrategy of(String fileType) {
+        try {
+            FileFormat fileFormat = FileFormat.valueOf(fileType.toUpperCase());
+            return fileFormat.getReadStrategy();
+        } catch (IllegalArgumentException e) {
+            log.warn("File source connector not support this file type [{}], it will be treated as a plain text file", fileType);
+            return new TextReadStrategy();
+        }
+    }
```

Comment:
When the user has no configuration we use the default txt, when the user has configuration but we do not support it should throw an error

Reply:
Resolved, pls review.